drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/rightpanel/ListGroupItemViewImpl.java

```diff
@@ -0,0 +1,146 @@
+
+    @DataField("listGroupItemContainer")
+    DivElement listGroupItemContainer = Document.get().createDivElement();
+
+    @DataField("faAngleRight")
+    SpanElement faAngleRight = Document.get().createSpanElement();
+
+    @DataField("fullClassName")
+    DivElement fullClassName = Document.get().createDivElement();
+
+    @DataField("factProperties")
+    UListElement factProperties = Document.get().createULElement();
+
+    Presenter presenter;
+
+    String factName;
+
+    String factType;
+
+    boolean toExpand = false;
+
+    @Override
+    public Widget asWidget() {
+        return null;
+    }
+
+    @EventHandler("listGroupItemHeader")
+    public void onListGroupItemHeaderClick(ClickEvent event) {
+        presenter.toggleRowExpansion(this, listGroupItemHeader.getClassName().contains("list-view-pf-expand-active"));
+    }
+
+    @Override
+    public void init(Presenter presenter) {
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void setToExpand(boolean toExpand) {
+        this.toExpand = toExpand;
+    }
+
+    @Override
+    public boolean isToExpand() {
+        return toExpand;
+    }
+
+    @Override
+    public void setFactName(String factName) {
+        this.factName = factName;
+        this.factType = factName;
+        fullClassName.setInnerText(factName);
+        listGroupItem.setAttribute("id", "listGroupItem-" + factName);
+    }
+
+    @Override
+    public void setFactNameAndType(String factName, String factType) {
+        this.factName = factName;
+        this.factType = factType;
+        String innerHtml = new StringBuilder()
+                .append("<b>")
+                .append(factName)
+                .append("</b> ")
+                .append(factType)
+                .toString();
+        fullClassName.setInnerHTML(innerHtml);
+        listGroupItem.setAttribute("id", "listGroupItem-" + factName);
+    }
+
+    @Override
+    public String getFactName() {
+        return factName;
+    }
+
+    @Override
+    public String getFactType() {
+        return factType;
+    }
+
+    @Override
+    public void addFactField(LIElement fieldElement) {
+        factProperties.appendChild(fieldElement);
+    }
+
+    @Override
+    public void addExpandableFactField(DivElement fieldElement) {
+        listGroupItemContainer.appendChild(fieldElement);
+    }
+
+    @Override
+    public DivElement getDivElement() {
+        return listGroupItem;
+    }
+
+    @Override
+    public void closeRow() {
+        listGroupItemHeader.removeClassName("list-view-pf-expand-active");
+        listGroupItemContainer.addClassName("hidden");
+        faAngleRight.removeClassName("fa-angle-down");
+    }
+
+    @Override
+    public void expandRow() {
+        listGroupItemHeader.addClassName("list-view-pf-expand-active");
+        listGroupItemContainer.removeClassName("hidden");
+        faAngleRight.addClassName("fa-angle-down");
```

Comment:
I think we could introduce constants for: "list-view-pf-expand-active", "hidden" and "fa-angle-down".

Reply:
Fixed with
[DROOLS-2909] Moved repeated String to constants